src/lib/conventional-commits.ts

```diff
@@ -181,17 +183,25 @@ export default function createConventionalCommits() {
       const message = serialize(commitMessage);
       output.appendLine(`message: ${message}`);
 
-      // 6. switch to scm and put message into message box
-      vscode.commands.executeCommand('workbench.view.scm');
-      repository.inputBox.value = message;
-      output.appendLine(`inputBox.value: ${repository.inputBox.value}`);
+      // 6. switch to scm and put message into message box or show the entire commit message in a separate tab
+      const showEditor = configuration.get<boolean>('showEditor');
+
+      if (showEditor) {
+        repository.inputBox.value = message;
+        openMessageInTab(repository);
+        output.appendLine(`show full commit message in a separate tab`);
+      } else {
+        vscode.commands.executeCommand('workbench.view.scm');
+        repository.inputBox.value = message;
+        output.appendLine(`inputBox.value: ${repository.inputBox.value}`);
+      }
 
       // 7. auto commit
       const autoCommit = configuration.get<boolean>('autoCommit');
-      if (autoCommit) {
+      if (autoCommit && !showEditor) {
         await vscode.commands.executeCommand('git.commit', repository);
+        output.appendLine('Finished successfully.');
```

Comment:
Why do we put `Finished successfully.` into `if (autoCommit && !showEditor)` condition?

Reply:
"Finished successfully." is now printed in two places - when the editor window is closed and here. If the print statement was not inside the if block, it would erroneously show _while_ the editor was open (ie, while the commit process was not finished yet).